### ISSUE TITLE
bug: eslint complains `The file does not match your project config: src/env/client.mjs (& others)` out of the box

### ISSUE THREAD

**Issue by timfee** [t3-oss/create-t3-app#342]
### Provide environment information

```
  System:
    OS: macOS 12.5.1
    CPU: (8) arm64 Apple M1 Pro
    Memory: 290.83 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.17.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 8.15.0 - /usr/local/bin/npm
```

### Describe the bug

Using project defaults, I get eslint errors out of the box on `mjs` and `cjs` files. 

It goes away when I add those extensions to `tsconfig.json`
```ts
  "include": [
    "next-auth.d.ts",
    "**/*.ts",
    "**/*.tsx",
    "**/*.js",
    "**/*.mjs", // <-- add
    "**/*.cjs"  // <-- add
  ],
```

Just noticed it today, so wanted to flag. Thanks!

### To reproduce

1. Run `yarn create t3-app` with TypeScript, tailwind and trpc:
   <img width="599" alt="Screenshot_2022-08-21_17-08-40_KLAbg6wX" src="https://user-images.githubusercontent.com/3246342/185817382-be5e5933-0c12-4b4b-8708-721b1d1c34db.png"> 
2. Open `src/env/client.mjs` and observe:
    <img width="869" alt="Screenshot_2022-08-21_17-08-13_dKL4RK9m" src="https://user-images.githubusercontent.com/3246342/185817477-bee9abda-544a-41f9-965a-b0d3442bb201.png">

**Comment by juliusmarminge** [comment]
Looks like you might have used an older version. `.mjs` files are included by default. Will change the `.js` to `.cjs` though. Regular `.js` files shouldn't exist.

**Comment by juliusmarminge** [comment]
Closed by #345
